platform/import-export/src/controllers/ImportJSONController.js

```diff
@@ -0,0 +1,94 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2017, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+define(
+    ['zepto'],
+    function ($) {
+
+        /**
+         * Controller for the `importJSONbutton` control type. Provides
+         * structure for a button (embedded via the template) which
+         * opens a filepicker and validates that the chosen file is legal
+         * JSON exported by Open MCT.
+         * @memberof platform/forms
+         * @constructor
+         * @param $scope the control's Angular scope
+         */
+        function ImportJSONController($scope, fileInputService) {
```

Comment:
It seems to me that this controller and the importjsondirective should be merged; you can put everything in the directive.  (Directives can depend on the fileInputService).